Add enterprise whitepaper (#1117)
pulumi/pulumi-hugo: themes/default/content/enterprise/whitepaper.md

---
title: Accelerating Enterprise Innovation through Cloud Engineering
description: |
    Pulumi is a cloud engineering platform that enables enterprises to use an unified software engineering process to deliver infrastructure and applications together for greater innovation velocity. 
meta_desc: |

>     Pulumi is a cloud engineering platform that enables enterprises to use an unified software engineering process to deliver infrastructure and applications together for greater innovation velocity. 

Reviewer: ```suggestion
    Pulumi is a cloud engineering platform that enables enterprises to use a unified engineering process to deliver infrastructure and applications together.
```

>     Pulumi is a cloud engineering platform that enables enterprises to use an unified software engineering process to deliver infrastructure and applications together for greater innovation velocity. 

Reviewer 2: ```suggestion
    Pulumi is a cloud engineering platform that gives enterprises a unified process for delivering infrastructure and applications for greater innovation velocity.
meta_desc: |
    Pulumi is a cloud engineering platform that gives enterprises a unified process for delivering infrastructure and applications for greater innovation velocity.
```

Reviewer 2: 159! 🤞

Reviewer 2: Ah @susanev looks like you committed just before I suggested.

Reviewer: oh i was just trying to get @aaronkao a preview, we can use urs!

Instead of discarding all the code written with the old tool, Mike used [Pulumi's conversion tool]({{< relref "/tf2pulumi" >}}) to assist with the transition. Mike says, “I knew that Pulumi provided an automatic conversion tool, and that helped a lot. I spent one day using the automatic conversion tool and then just spent some time making sure everything looked right. In two days, I was all set.”

As an example of how Pulumi made formerly complex tasks straightforward, Mike says, “With the old tool, spinning up our databases meant we had to have about 20 blocks of code and do a lot of copy and pasting. But with Pulumi, it's Python. So you have a for statement and then you define the names of the databases you want. It's five lines of code. If you want to add a new database, add one line, and you're good to go.”

> Pulumi also simplified cloud infrastructure management and provided clear visibility into all active resources. Mike said, “Before Pulumi, we had a manager for infrastructure state files that was a complete pain. We ran into weird issues and errors, and it was difficult to ask questions, such as 'How many developer instances are running?' Now, Pulumi manages all the state files for us and gives us that visibility.""

Reviewer 2: ```suggestion
Pulumi also simplified cloud infrastructure management and provided clear visibility into all active resources. Mike said, “Before Pulumi, we had a manager for infrastructure state files that was a complete pain. We ran into weird issues and errors, and it was difficult to ask questions, such as 'How many developer instances are running?' Now, Pulumi manages all the state files for us and gives us that visibility."
```

## Improving Developer Productivity with Self-Service Cloud Infrastructure

Shortly afterward, Mike reviewed a survey which asked Bitbucket developers how they spent their time. He was surprised to learn that developers were spending 20% of their workweek maintaining their private Bitbucket development environments.

Again, part of the problem rested with the old tool, whose DSL was unfamiliar to most developers. Because it lacked standard programming capabilities, it made the infrastructure code verbose and difficult to understand. Mike knew that he could streamline the entire workflow with Pulumi and make infrastructure more accessible and simpler to use for developers who understand Python.

To make things simple and turnkey for developers, the DevSpeed team built a self-service dashboard that created and configured new Bitbucket instances. Developers enter a few pieces of information, such as what they want to call the instance, what region to use, and an SSH key. Then, a CI/CD pipeline using Bitbucket Pipelines and Bamboo calls Pulumi to deploy a complete developer environment.

> Once the instance is ready, the developer simply SSH’s into it, and the instance it will look exactly like Bitbucket. The developer can make the changes they want and then create a pull request to merge their work into a main branch. The developer can also share the URL of the instance with teammates so that they can see the changes when they review the code.

Reviewer 2: ```suggestion
Once the instance is ready, the developer simply SSH’s into it, and the instance will look exactly like Bitbucket. The developer can make the changes they want and then create a pull request to merge their work into a main branch. The developer can also share the URL of the instance with teammates so that they can see the changes when they review the code.
```

### Results

Using Pulumi with its existing CI/CD process, the DevSpeed team built a self-service dashboard that enables any Bitbucket developer to quickly and easily provision a cloud-based development environment. Compared to the DSL-based tool they previously used, Pulumi significantly reduced the number of lines of code needed to model infrastructure and made the code much easier to read and understand. In terms of user satisfaction, the new dashboard meant that developers went from spending one day a week maintaining their environment to less than four hours per week. Mike’s team also receives significantly fewer infrastructure questions from developers on their Slack help channel.

## Summary

Here is a summary of how Pulumi helped Bitbucket and the DevSpeed team.

- Allowed developers to use general purpose programming languages like Python and familiar software tools to build and manage infrastructure as code
- Enabled Bitbucket developers to easily provision approved cloud infrastructure using a self-service dashboard built with Pulumi

> - Reduced the time Bitbucket developers spent maintaining their instances from 8 hours per week to fewer than four.

Reviewer 2: ```suggestion
- Reduced the time Bitbucket developers spent maintaining their instances from 8 hours per week to fewer than four
```

- Boosted productivity of developers in Sydney by extending cloud deployments into another AWS region using Infrastructure as Code in Python

> - Reduced the size and complexity of codebase for managing infrastructure while increasing its clarity

Reviewer 2: ```suggestion
- Reduced the size and complexity of codebase for managing infrastructure while increasing its clarity

```

meta_desc: |
    Pulumi is a cloud engineering platform that enables enterprises to use an unified software engineering process to deliver infrastructure and applications together for greater innovation velocity. 
    
type: case-studies

customer_name: Enterprise Whitepaper
customer_logo: 
customer_url: 

exec_summary: |
    Cloud engineering is the practice of applying software engineering practices and tools uniformly across infrastructure, development, and compliance teams to tame the complexity of delivering and managing modern cloud applications. Pulumi is a cloud engineering platform that enables enterprises to use an unified software engineering process to deliver infrastructure and applications together at greater scale and velocity. Pulumi is a trusted advisor and partner for enterprises as they embark on their cloud engineering journey. Organizations from high growth startups to the Global 2000 have chosen Pulumi to tame cloud complexity, reduce risk through automation, increase reliability and security, and accelerate development velocity. 

> 
> 

Reviewer 2: ```suggestion
```

To make things simple and turnkey for developers, the DevSpeed team built a self-service dashboard that created and configured new Bitbucket instances. Developers enter a few pieces of information, such as what they want to call the instance, what region to use, and an SSH key. Then, a CI/CD pipeline using Bitbucket Pipelines and Bamboo calls Pulumi to deploy a complete developer environment.

Once the instance is ready, the developer simply SSH’s into it, and the instance it will look exactly like Bitbucket. The developer can make the changes they want and then create a pull request to merge their work into a main branch. The developer can also share the URL of the instance with teammates so that they can see the changes when they review the code.

Below is a diagram showing how a Bitbucket cloud instance is architected:

> <img class="block mx-auto md:max-w-4xl my-8"
> src="/images/case-studies/bitbucket-cloud-diagram.png" alt="Bitbucket Cloud and Pulumi">

Reviewer 2: ```suggestion
<img class="block mx-auto md:max-w-4xl my-8" src="/images/case-studies/bitbucket-cloud-diagram.png" alt="Bitbucket Cloud and Pulumi">
```
